packages/ui/__tests__/components/organisms/DropdownCardSwitch.test.js

```diff
@@ -0,0 +1,117 @@
+import {shallow} from 'enzyme';
+import {DropdownCardSwitch, DropdownCard, Text} from '@axelor/aos-mobile-ui';
+import {getGlobalStyles} from '../../tools';
+
+describe('DropdownCardSwitch', () => {
+  const dropdownItems = [
+    {key: 1, title: 'Item 1', childrenComp: <Text>Content 1</Text>},
+    {key: 2, title: 'Item 2', childrenComp: <Text>Content 2</Text>},
+  ];
+
+  it('renders without crashing', () => {
+    const wrapper = shallow(
+      <DropdownCardSwitch dropdownItems={dropdownItems} />,
+    );
+
+    expect(wrapper.exists()).toBe(true);
+  });
+
+  it('renders correctly with dropdown items', () => {
+    const wrapper = shallow(
+      <DropdownCardSwitch dropdownItems={dropdownItems} />,
+    );
+    expect(wrapper.find(DropdownCard)).toHaveLength(dropdownItems.length);
+  });
+
+  it('toggles dropdown card open and close on press', () => {
+    const wrapper = shallow(
+      <DropdownCardSwitch dropdownItems={dropdownItems} />,
+    );
+
+    wrapper.find(DropdownCard).first().simulate('press');
+    expect(wrapper.find(DropdownCard).first().props().dropdownIsOpen).toBe(
+      true,
+    );
+
+    wrapper.find(DropdownCard).first().simulate('press');
+    expect(wrapper.find(DropdownCard).first().props().dropdownIsOpen).toBe(
+      false,
+    );
+  });
+
+  it('shows children component for the opened card', () => {
+    const wrapper = shallow(
+      <DropdownCardSwitch dropdownItems={dropdownItems} />,
+    );
+
+    expect(wrapper.find(DropdownCard).first().prop('dropdownIsOpen')).toBe(
+      false,
+    );
+
+    wrapper.find(DropdownCard).first().simulate('press');
+
+    expect(wrapper.find(DropdownCard).first().prop('dropdownIsOpen')).toBe(
+      true,
+    );
+
+    wrapper.find(DropdownCard).first().simulate('press');
+
+    expect(wrapper.find(DropdownCard).first().prop('dropdownIsOpen')).toBe(
+      false,
+    );
+  });
+
+  it('applies custom style when provided', () => {
+    const customStyle = {width: 200};
+    const wrapper = shallow(
+      <DropdownCardSwitch dropdownItems={dropdownItems} style={customStyle} />,
+    );
+
+    expect(getGlobalStyles(wrapper.find(View))).toMatchObject(customStyle);
+  });
+
+  it('applies custom styleTitle when provided', () => {
+    const customStyleTitle = {fontWeight: 'bold'};
+    const wrapper = shallow(
+      <DropdownCardSwitch
+        dropdownItems={dropdownItems}
+        styleTitle={customStyleTitle}
+      />,
+    );
+
+    wrapper.find(DropdownCard).forEach(dropdownCard => {
+      expect(dropdownCard.prop('styleText')).toEqual(customStyleTitle);
+    });
+  });
+
+  it('passes the correct title DropdownCard', () => {
+    const titleProps = 'Test';
+    const wrapper = shallow(
+      <DropdownCardSwitch dropdownItems={dropdownItems} title={titleProps} />,
+    );
+    dropdownItems.forEach((item, index) => {
+      const dropdownCard = wrapper.find(DropdownCard).at(index);
+      expect(dropdownCard.prop('title')).toEqual(item.title);
+    });
```

Comment:
Should go to 'renders correctly with dropdown items'
titleProps is useless here, DropdownCardSwitch does not have a title props.